netutils/webclient/webclient.c

```diff
@@ -674,225 +693,336 @@ int webclient_perform(FAR struct webclient_context *ctx)
 
   /* Initialize the state structure */
 
-  ws = calloc(1, sizeof(struct wget_s));
-  if (!ws)
+  if (ctx->ws == NULL)
     {
-      return -errno;
-    }
+      ws = calloc(1, sizeof(struct wget_s));
+      if (!ws)
+        {
+          return -errno;
+        }
 
-  ws->buffer = ctx->buffer;
-  ws->buflen = ctx->buflen;
+      ws->buffer = ctx->buffer;
+      ws->buflen = ctx->buflen;
 
-  /* Parse the hostname (with optional port number) and filename
-   * from the URL.
-   */
+      /* Parse the hostname (with optional port number) and filename
+       * from the URL.
+       */
 
-  ret = parseurl(ctx->url, ws);
-  if (ret != 0)
-    {
-      nwarn("WARNING: Malformed URL: %s\n", ctx->url);
-      free(ws);
-      return ret;
+      ret = parseurl(ctx->url, ws);
+      if (ret != 0)
+        {
+          nwarn("WARNING: Malformed URL: %s\n", ctx->url);
+          free(ws);
```

Comment:
```suggestion
          free(ws);
          ctx->ws = NULL;
```
To avoid use-after-free errors, the freed pointer should be nullified.
```

Reply:
This is one is not actually an issue since it `ctx->ws` is already NULL due to checking on line 696.